### ISSUE TITLE
Trying to load new content into player item crashes.

### ISSUE THREAD

**Issue by russelltwarwick** [DZamataev/DZVideoPlayerViewController#5]
I am trying to load a new video into the player. The issue is that when the current video is loading it crashes when you try load in a new video over that. 
Please help
`'Cannot remove an observer <DZVideoPlayerViewController 0x7faf1600e800> for the key path "status" from <AVPlayerItem 0x608000002030> because it is not registered as an observer.'`

**Comment by DZamataev** [comment]
Have you tried calling
```
- (void)stop;
```
on the controller and then waiting for
```
- (void)playerDidStop;
```
from the delegate before starting new playback?

**Comment by russelltwarwick** [comment]
Yes this doesn't seem to work. Seems to be fine when trying to load a new live stream link in but has issues when it comes to mp4

**Comment by russelltwarwick** [comment]
It is only and issue when you try load a new mp4 while the other mp4 is still loading

**Comment by DZamataev** [comment]
Ok, now I see. That might be really easy to fix.
Find the following line in DZVideoPlayerViewController class
```
        [self.playerItem removeObserver:self forKeyPath:@"status" context:&ItemStatusContext];
```
and try catch it like so
```
        @try {
            [self.playerItem removeObserver:self forKeyPath:@"status" context:&ItemStatusContext];
        }
        @catch(id anException) {
            //do nothing
        }
```

**Comment by russelltwarwick** [comment]
Seems to work but as soon as it starts playing it crashes again.
only seems to be delaying the same issue

**Comment by russelltwarwick** [comment]
`Key path: status, Options: <New: YES, Old: NO, Prior: NO> Context: 0x10e9abb90`

**Comment by DZamataev** [comment]
Which line is causing the crash?

**Comment by russelltwarwick** [comment]
Seemed to have got this working now. Thanks for your help :) 
One last question though.
When leaving a view controller. 
What is the best way to deallocate the video player?

**Comment by DZamataev** [comment]
Why would you need it? I suppose everything gets deallocated with the View Controller? Am I missing something?

**Comment by kuldeep13990** [comment]
@DZamataev , i put removeObserver in @try @catch block but still app is getting crashed with below error.

*** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'An instance 0x170013cb0 of class AVPlayerItem was deallocated while key value observers were still registered with it. Current observation info: <NSKeyValueObservationInfo 0x17042a280> (
<NSKeyValueObservance 0x170651e20: Observer: 0x1268d2a00, Key path: status, Options: <New: YES, Old: NO, Prior: NO> Context: 0x1006bd600, Property: 0x17425ddc0>
)'
